src/graph_synchronizer.cpp

```diff
@@ -122,37 +119,36 @@ void GraphSynchronizer::Lock::lock() {
             // Make them into pos_ts that point left to right, the way Jordan thinks.
             pos_t left_pos = make_pos_t(start_left.node, start_left.is_end, 0);
             pos_t right_pos = make_pos_t(end_right.node, !end_right.is_end,
-                synchronizer.graph.get_node(end_right.node)->sequence().size() - 1);
+                synchronizer.graph.get_node(end_right.node)->sequence().size());
             
             // Since these are already at node ends, we don't need to worry about node cuts.
             
             // Extract paths out to the length we need to connect the ends, or a bit further.
             // TODO: be sure to extract really big indels somehow...
```

Comment:
@adamnovak 
I could use a little review on this. 

I eliminated one of the options you were using here because I decided it wasn't good. Before you could request that the source and destination positions would be the only tips in the extracted graph. I got rid of that option. You do still have the option of requiring that all material in the subgraph is reachable from the source position and can reach the destination position, which is a slightly stronger property. I switched you over to that for now.

I also no longer provide the guarantee that the node IDs will be the same as long as you don't duplicate. Also take a look out my translation solution to that mismatch.

Reply:
I guess that's sufficient to pass whatever tests we have in there for vg add.

I'd have to go back and think about how vg add thinks to really evaluate what it means that the node IDs might not be "the same" anymore. I can proibably look at this tomorrow.

But the reachability criterion ought to be good enough for vg add; I don't think it is going to be trying to add lots of ugly unpaired reversing edges.

Reply:
I wasn't super clear about this, but I believe the node translation trick I do after the graph extraction should make the node ID space the same as the full graph again.